### PR TITLE
Update Pillar styles

### PR DESCRIPTION
**Pull request by josepmartins**
## Summary

Updates `Pillar` icon size, spacing and font settings.

Fixes: https://github.com/github/primer/issues/2505

## List of notable changes:

- **updated** icon size to be `medium` (32px) 
- **updated** heading font settings to use `size 6` (20px/14px)
- **updated** spacing between icon and heading, heading and description, description and link
- **updated** link spacing to use `margin-top: auto`. This property adjust the position depending on it's sibling height size.
- **added** stacked with link story

## Steps to test:

Storybook 👀 : https://primer-865de34718-26139705.drafts.github.io/storybook/?path=/story/components-pillar-features--stacked-with-link

## Supporting resources (related issues, external links, etc):

- https://github.com/github/primer/issues/2505

## Contributor checklist:

- [ ] All new and existing CI checks pass
- [ ] Tests prove that the feature works and covers both happy and unhappy paths
- [ ] Any drop in coverage, breaking changes or regressions have been documented above
- [ ] New visual snapshots have been generated / updated for any UI changes
- [ ] All developer debugging and non-functional logging has been removed
- [ ] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

## Screenshots:

> Please try to provide before and after screenshots or videos

<table>
<tr>
<th> Before</th> <th> After </th>
</tr>
<tr>
<td valign="top">

Screenshot 2023-08-03 at 13 40 46

 </td>
<td valign="top">

Screenshot 2023-08-03 at 13 41 13

</td>
</tr>
</table>

### CODE REVIEW

File: packages/react/src/Pillar/Pillar.tsx [hunk 1]
```diff
@@ -72,7 +72,7 @@ function PillarIcon({icon: Icon, className, color = defaultPillarIconColor, ...r
   return (
     <span className={clsx(styles.Pillar__icon, styles[`Pillar__icon--color-${color}`], className)} {...rest}>
       {typeof Icon === 'function' ? (
-        <Icon size={24} />
+        <Icon size={'medium'} />
```
Comment (danielguillan) [inline packages/react/src/Pillar/Pillar.tsx:75]:
```suggestion
        <Icon size="medium" />
```

**Comment by josepmartins** [comment]
cc @jesussandreas @ajashams @nsolerieu in case you want to take a look:


Storybook: https://primer-865de34718-26139705.drafts.github.io/storybook/?path=/story/components-pillar-features--stacked-with-link

Docs: https://primer-865de34718-26139705.drafts.github.io/components/Pillar/react

File: packages/react/src/Pillar/Pillar.features.stories.tsx [hunk 2]
```diff
@@ -176,3 +176,20 @@ export const StackedCentered: StoryFn<typeof Pillar> = () => {
     </Stack>
   )
 }
+
+export const StackedWithLink: StoryFn<typeof Pillar> = () => {
+  return (
+    <Stack direction="horizontal" gap={'spacious'} padding={'spacious'}>
```
Comment (rezrah) [inline packages/react/src/Pillar/Pillar.features.stories.tsx:182]:
```suggestion
    <Stack direction="horizontal" gap="spacious" padding="spacious">
```

**Review by rezrah — APPROVED** [review]
👍 Looks good.

Could you add a changeset before you merge it please @josepmartins?

File: packages/react/src/Pillar/Pillar.features.stories.tsx (hunk 2 above)
Comment (josepmartins) [reply packages/react/src/Pillar/Pillar.features.stories.tsx]: done!

**Pull request merged**